### PR TITLE
Earlier IP anonymization

### LINKED ISSUE #119: Respect `team.anonymize_ips` earlier in the pipeline and hide IPs from plugins
We store the IP address of the user calling `posthog.capture` inside the `$ip` property. If this property exists, we use it. If not, we take the IP from the request and put it in the property.

When the event passes through plugins, we pass along the request's IP, all the way until ingestion. If a plugin wants to export the event to another database, it'll also have access to and send the request's IP, not the provided `$ip` property.

Solution: we should override the IP earlier in the pipeline, as to prevent storing PII that we don't want.

### PR DESCRIPTION
**Pull request by mariusandra**
## Changes

- Under "Project" replace the checkbox "Anonymize client IP data" with "Discard client IP data", which better reflects what this checkbox does. "Anonymize" could also mean e.g. removing 8 random bits.
- Actually hides the IP before the task queues and plugins if the option is flipped, instead of just not storing it in the final ingestion stage
- Resolves https://github.com/PostHog/plugin-server/issues/119

This came out of the needs of a client who wants to use plugins to send events to BigQuery, yet found the IP stored even though they had the flag set.

Draft because:
- Requires corresponding changes to `plugin-server` and `plugin-scaffold`
- I want to test that nothing breaks

Misc question:
- Should we remove the check for `team.anonymize_ips` inside the process event functions after the queues?
  - Pros: gets rid of legacy code. 
  - Cons: there might be an edge case where an IP slips through when updating posthog for in-flight events

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

### CODE REVIEW

File: frontend/src/scenes/project/Settings/IPCapture.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ export function IPCapture() {
                     marginLeft: '10px',
                 }}
             >
-                Anonymize client IP data
+                Do not save client IP data
```
Comment (Twixes) [inline frontend/src/scenes/project/Settings/IPCapture.js:23]: How bout "Discard"?

Comment (mariusandra) [reply frontend/src/scenes/project/Settings/IPCapture.js:23]: Could work. "Discard client IP addresses" or "... IP data"?

**Comment by mariusandra** [comment]
I added a test and also tested manually for both EE and OSS pipelines and both ingestions (python & js). It seems to work.

**Review by Twixes — APPROVED** [review]
Makes sense to me

**Pull request merged**